middleware/common/source/serialize/toml.cpp

```diff
@@ -0,0 +1,388 @@
+//! 
+//! Copyright (c) 2015, The casual project
+//!
+//! This software is licensed under the MIT license, https://opensource.org/licenses/MIT
+//!
+
+
+#include "common/serialize/toml.h"
+
+#include "common/serialize/create.h"
+
+#include "common/transcode.h"
+#include "common/buffer/type.h"
+
+#include "common/code/raise.h"
+#include "common/code/casual.h"
+
+#include <toml++/toml.hpp>
+
+namespace casual
+{
+   namespace common
+   {
+      namespace serialize
+      {
+         namespace toml
```

Comment:
I think you should collapse these to `common::serialise::toml` 
And do the explicit template instantiation at the end as ` template struct  common::serialize::reader::create::Registration< toml::local::reader::Implementation>:` and so on.